base/collection.go

```diff
@@ -532,7 +532,11 @@ func (c *Collection) Incr(k string, amt, def uint64, exp uint32) (uint64, error)
 }
 
 func (c *Collection) StartDCPFeed(args sgbucket.FeedArguments, callback sgbucket.FeedEventCallbackFunc, dbStats *expvar.Map) error {
-	return StartDCPFeed(c, c.Spec, args, callback, dbStats)
+	if !IsEnterpriseEdition() {
+		// use new DCP feed in CBG-2232
+		return StartDCPFeed(c, c.Spec, args, callback, dbStats)
+	}
```

Comment:
This does not feel like the right place to decide whether to use cbdatasource or gocb...

We _always_ want to use gocb for the mutation/changes feed, enterprise or not.

Can `StartImportFeed` skip calling `collection.StartDCPFeed` and instead call `Start(Cbdatasource)DCPFeed` directly? We have everything available up one level (including the EE/CE check), and import feels like the special case here.

![Screenshot 2022-08-02 at 14 09 52](https://user-images.githubusercontent.com/1525809/182382600-4a93b975-052b-475a-9e79-57bc1ce4f868.png)

Reply:
In fact, I can't think of a reason why we'd need to use cbdatasource at all - community edition will still support collections (right?), so we have to use gocb for its import feed too. The decision of which feed to use for import "sharded or not sharded", not "gocb or cbdatasource" - could that help simplify this code?

Reply:
Moved this up a level.